src/local/butler/reproduce.py

```diff
@@ -220,24 +226,27 @@ def _reproduce_crash(testcase_url, build_directory):
   result = testcase_manager.test_for_crash_with_retries(testcase, testcase_path,
                                                         timeout)
 
+  # Get the return code and symbolized stacktrace before cleaning up.
+  return_value = (result.is_crash(), result.get_stacktrace())
```

Comment:
Instead of this, why not move the remove_directory in a cleanup function at very end. dont like tuples being used here.

Reply:
The main issue is that _reproduce_crash is the entry point for tests, and I don't like the idea of having to do additional cleanup after calling this.

Reply:
Fine as-is then.

Reply:
I think I might see a way around it. I'll try to address it if possible but if I can't find a good solution I'll skip and land it as-is.

Reply:
Discussed some of this offline.

Looks like the conversion to pyfakefs is going to be a bit more involved than I thought. I'm going to move this to a TODO and change it back later, but leave it as-is for now.